client/src/pages/Login/Login.tsx

```diff
@@ -1,53 +1,81 @@
 import { AiOutlineMail } from "react-icons/ai";
-import { Input } from "./Input/Input";
 import { RiLockPasswordFill } from "react-icons/ri";
-import { Button } from "../../components/Button/Button";
-
-import { useInput } from "./useInput";
-import { useMutation } from "../../hooks/useMutation";
-
+import { IoLanguageSharp } from "react-icons/io5";
+import { useInput } from "../../components/Input/useInput";
 import "./login.css";
-import { ajaxClient } from "../../ajaxClient";
+import { Input } from "../../components/Input/Input";
+import { ajaxClient } from "ajaxClient";
+import { useMutation } from "hooks/useMutation";
+import { Button } from "components/Button/Button";
+import { delayTask } from "utils";
 
 type LoginProps = {
-  onLoginComplete: (id: string) => void;
+  onAuthComplete: (id: string) => void;
+};
+
+const options = {
+  signup: {
+    title: "Sign Up",
+    url: "/users",
+    submitButton: "Sign up",
+    switchButton: "Already have an account? Login",
+    loadingMessage: "Creating new account...",
+    successMessage: "Your account is created...",
+  },
+  login: {
+    title: "Login",
+    url: "/auth/login",
+    submitButton: "Log in",
+    switchButton: "Create New Account",
+    loadingMessage: "Validating credentials...",
+    successMessage: "Login validated...",
+  },
 };
 
 export const Login = (props: LoginProps) => {
   const [id, handleIDChange] = useInput("");
   const [password, handlePasswordChange] = useInput("");
+  const [name, handleNameChange] = useInput("");
+  const [mode, setMode] = useState<"login" | "signup">("login");
 
-  const validateLogin = useCallback(
-    () => ajaxClient.post("/auth/login", { id, password }),
-    [id, password]
+  const switchMode = () => setMode((m) => (m === "login" ? "signup" : "login"));
+
+  const validateAuth = useCallback(
+    (id, password, name) => {
+      let data = {};
+      data = { id, password };
+      if (mode === "signup") data = { user: { id, password, name } };
+      return ajaxClient.post(options[mode].url, data);
+    },
+    [mode]
   );
 
-  const mutation = useMutation<boolean>(validateLogin);
+  const mutation = useMutation<boolean>(validateAuth);
 
-  let loginStatus = "";
-  if (mutation.status === "loading") loginStatus = "Verifying credentials...";
+  let status = "";
+  if (mutation.status === "loading") status = options[mode].loadingMessage;
   else if (mutation.status === "error") {
-    loginStatus = mutation.error?.message || "Some error occured...";
-  } else if (mutation.status === "success") {
-    loginStatus = "Login validated...";
-    setTimeout(() => {
-      props.onLoginComplete(id);
-    }, 300);
-  }
+    status = mutation.error?.message || "Some error occured...";
+  } else if (mutation.status === "success")
+    status = options[mode].successMessage;
+
+  const { onAuthComplete } = props;
+  useEffect(() => {
+    if (mutation.status === "success") {
+      delayTask(() => onAuthComplete(id), 0.3);
```

Comment:
Take success and failure handlers in useMutation hook itself instead of such side effects each time